API.md

```diff
@@ -0,0 +1,96 @@
+### StorageBuckets API Interface
+
+#### The StorageBucketManager Interface
+```
+[
+  Exposed=(Window,Worker),
+  SecureContext
+] interface StorageBucketManager {
+    [NewObject] Promise<StorageBucket> openOrCreate(USVString name,
+                                                    optional StorageBucketOptions options = {});
+    Promise<sequence<USVString>> keys();
+    Promise<undefined> delete(USVString name);
+};
+
+dictionary StorageBucketOptions {
+  USVString? title = null,
+  boolean persisted = false,
+  StorageBucketDurability durability = "relaxed",
+  unsigned long long? quota = null,
+  DOMTimeStamp? expires = null,
+}
+
+enum StorageBucketDurability {
+  "strict",
+  "relaxed"
+}
+
+[SecureContext] interface StorageBucket {
+  [Exposed=Window] Promise<boolean> persist();
+  Promise<boolean> persisted();
+
+  Promise<StorageEstimate> estimate();
+
+  Promise<StorageBucketDurability> durability();
+
+  Promise<undefined> setExpires(DOMTimeStamp expires);
+  Promise<DOMTimeStamp> expires();
+}
+```
+
+### navigator.storageBuckets
+```
+partial interface Navigator {
+  [SecureContext, SameObject] readonly attribute StorageBucketManager storageBuckets;
+}
+```
+
+### Indexed DB
+```
+partial interface StorageBucket {
+  [SameObject] readonly attribute IDBFactory indexedDB;
+}
+```
+
+### Cache Storage
+```
+partial interface StorageBucket {
+  [SameObject] readonly attribute CacheStorage caches;
+}
+```
+
+### File System Access
+```
+partial interface StorageBucket {
+  promise<FileSystemDirectoryHandle> getDirectory();
+}
+```
+
+### Service Worker
+```
+interface mixin ServiceWorkerContainerRegistration {
+  [NewObject] Promise<ServiceWorkerRegistration> register(USVString scriptURL,
+                                                          optional RegistrationOptions options = {});
+  [NewObject] Promise<any> getRegistration(optional USVString clientURL = "");
+  [NewObject] Promise<FrozenArray<ServiceWorkerRegistration>> getRegistrations();
+}
+
+interface StorageBucketServiceWorkerContainer {}
+
+StorageBucketServiceWorkerContainer includes ServiceWorkerContainerRegistration;
+
+partial interface StorageBucket {
+  [SameObject] readonly attribute StorageBucketServiceWorkerContainer serviceWorker;
```

Comment:
I don't think this correct..but wasn't sure how to do this best. The intent was to expose only a subset of methods from [ServiceWorkerContainer](https://www.w3.org/TR/service-workers/#serviceworkercontainer) as detailed in the [explainer](https://github.com/WICG/storage-buckets/blob/gh-pages/explainer.md#storage-buckets-and-service-workers) so it would only have the following methods.
```
bucket.serviceWorker.register(...)
bucket.serviceWorker.getRegistration(...)
bucket.serviceWorker.getRegistrations(...)
```
Do you have any advice for how to do this best?

Reply:
Let's write down the intended goal at the beginning of the `Service Worker` section.

I think it's worth describing that `ServiceWorkerContainerRegistration` pulls some members from `ServiceWorkerContainer`, and we'd ultimately change the Service Workers spec.

(Also, let's make this the last section in the doc, so that people who get distracted by all the complexity here would have already read the integration with the File API.)

Reply:
Sounds good!